### PR TITLE
New blog for asp dotnet and react with docker

### PR DESCRIPTION
**Pull request by Luqmant51**
```
New blog for docker-compose with react.ts and asp-dot net
```

### CODE REVIEW

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
+---
+layout: distill
+title: Krazzy Adventures
+description: A booking website where users can book to participate in different events.
+tags: Krazzy booking
```
Comment (ZuhaibSohail163) [inline _posts/2024-14-18-asp-react-docker-compose.md:5]: change title and tags, date

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+  - name:  Sheikh Luqman
+    url: "https://en.wikipedia.org/wiki/Boris_Podolsky"
+    affiliations:
+      name: Allied Consultants
+  - name: Zuhaib Sohail
+    url: "https://github.com/ZuhaibSohail163"
+    affiliations:
+      name: Allied Consultants
+
+bibliography: 2018-12-22-krazzy.bib
```
Comment (ZuhaibSohail163) [inline _posts/2024-14-18-asp-react-docker-compose.md:24]: also this

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+# Optionally, you can add a table of contents to your post.
+# NOTES:
+#   - make sure that TOC names match the actual section names
+#     for hyperlinks within the post to work correctly.
+#   - we may want to automate TOC generation in the future using
+#     jekyll-toc plugin (https://github.com/toshimaru/jekyll-toc).
+toc:
```
Comment (ZuhaibSohail163) [inline _posts/2024-14-18-asp-react-docker-compose.md:32]: also this

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+    text-align: left;
+    margin: 12px 0;
+    text-align: center;
+    font-size: 16px;
+  }
+
+---
+
+ASP Dot Net 8.0 and React.ts with Docker compose
+Create main folder name like `asp-react-nginx-compose` or other suit able.
```
Comment (ZuhaibSohail163) [inline _posts/2024-14-18-asp-react-docker-compose.md:88]: check spelling

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+3. Open Docker from your Applications to complete the installation.
+4. Docker may request your password to install a helper tool; provide your password to continue.
+
+### For Linux
+
+The installation commands can vary depending on your Linux distribution. Here's how you can install Docker on Ubuntu:
+
+1. Update your existing list of packages:
+```bash
+   sudo apt update
```
Comment (ZuhaibSohail163) [inline _posts/2024-14-18-asp-react-docker-compose.md:130]: write a `$` sign before every command for convention.like
`$ sudo apt update`

**Review by ZuhaibSohail163 — COMMENTED** [review]
Please address the requested changes

File: _posts/2024-14-18-asp-react-docker-compose.md (hunk 2 above)
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:24]: removed

File: _posts/2024-14-18-asp-react-docker-compose.md (hunk 3 above)
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:32]: removed the table of content

File: _posts/2024-14-18-asp-react-docker-compose.md (hunk 4 above)
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:88]: checked

File: _posts/2024-14-18-asp-react-docker-compose.md (hunk 5 above)
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:130]:
```
Iplement as advised
```

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+  }
+
+---
+
+ASP Dot Net 8.0 and React.ts with Docker compose
+Create root folder name like `asp-react-nginx-compose` or other suit able.
+In it create 3 folders and 1 file in root folder :
+#### The Project structure is:
+```plaintext
+app
```
[the remark this answers is not in the document]
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:48]: done

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+4 directories, 1 file
+```
+
+### ASP.NET 8.0 and React.ts with Docker Compose
+
+This guide explains how to set up a project structure using ASP.NET, React, and Nginx within Docker containers. Below are detailed steps for configuring your development environment, including Docker installation.
+
+* For enabling Docker support in project it is important Docker is install and running in Laptop and computer.
+## Installing Docker
```
[the remark this answers is not in the document]
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:63]: done

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+4 directories, 1 file
+```
+Here is the reference of project public repo in github [Source Code](https://github.com/usman-jamil/dev-docs/tree/main) of given example project asp-react-nginx-compose.
+
+### ASP.NET 8.0 and React.ts with Docker Compose
+
+This guide explains how to set up a project structure using ASP.NET, React, and Nginx within Docker containers. Below are detailed steps for configuring your development environment, including Docker installation.
+
+* For enabling Docker support in project it is important Docker is install and running in Laptop and computer.
+## Installing Docker
```
Comment (ZuhaibSohail163) [inline _posts/2024-14-18-asp-react-docker-compose.md:63]: remove

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+```plaintext
+app
+├── backend
+├── docker-compose.yml
+├── frontend
+└── nginx
+
+4 directories, 1 file
+```
+Here is the reference of project public repo in github [Source Code](https://github.com/usman-jamil/dev-docs/tree/main) of given example project asp-react-nginx-compose.
```
Comment (ZuhaibSohail163) [inline _posts/2024-14-18-asp-react-docker-compose.md:56]: give correct link

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+<img src="../assets/img/Screenshot1.png" width="50%" height="50%">
+
+#### After Project Creation
+
+Right-click on the project name in your IDE, then:
+
+1. Click "Add"
+2. Select "Docker Support"
+
+<img src="../assets/img/Screenshot2.png" width="50%" height="50%">
```
Comment (usman-jamil) [inline _posts/2024-14-18-asp-react-docker-compose.md:121]: Remove these lines in favor of "please see the Dockerfile inside backend folder. It contains the following contents. Notice how the API is exposed on port 8080 and port 8081"

File: _posts/2024-14-18-asp-react-docker-compose.md (hunk 8 above)
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:63]: done

File: _posts/2024-14-18-asp-react-docker-compose.md (hunk 10 above)
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md]: done

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+
+4 directories, 1 file
+```
+Here is the reference of project public repo in github [Source Code](https://github.com/Luqmant51/asp-react-nginx-compose) of given example project asp-react-nginx-compose.
+
+### ASP.NET 8.0 and React.ts with Docker Compose
+
+This guide explains how to set up a project structure using ASP.NET, React, and Nginx within Docker containers. Below are detailed steps for configuring your development environment, including Docker installation.
+
+* For enabling Docker support in project it is important Docker is install and running in Laptop and computer.
```
Comment (usman-jamil) [inline _posts/2024-14-18-asp-react-docker-compose.md:62]: `installed and running` not `install and running`

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+├── frontend
+└── nginx
+
+4 directories, 1 file
+```
+Here is the reference of project public repo in github [Source Code](https://github.com/Luqmant51/asp-react-nginx-compose) of given example project asp-react-nginx-compose.
+
+### ASP.NET 8.0 and React.ts with Docker Compose
+
+This guide explains how to set up a project structure using ASP.NET, React, and Nginx within Docker containers. Below are detailed steps for configuring your development environment, including Docker installation.
```
Comment (usman-jamil) [inline _posts/2024-14-18-asp-react-docker-compose.md:60]: remove` , including Docker installation`

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 13]
```diff
@@ -0,0 +1,250 @@
...
+
+* For enabling Docker support in project it is important Docker is install and running in Laptop and computer.
+Before you start setting up your project, you need to install Docker on your machine if it isn't already installed. Docker allows you to create, deploy, and run applications by using containers.
```
Comment (usman-jamil) [inline _posts/2024-14-18-asp-react-docker-compose.md:63]: Add: We will see how `docker compose` can be used to perform orchestration between containers, keeping isolation in mind i.e., while some containers can talk to each other, we might not want that for others e.g., have NGINX be the communications broker between the front-end and back-end and not allow them to talk to each other directly. This has several use-cases which are outside the scope of this blog.

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 14]
```diff
@@ -0,0 +1,250 @@
...
+  $ sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+
+  $ sudo apt update
+
+  $ apt-cache policy docker-ce
+
+  $ sudo apt install docker-ce
+
+docker run hello-world
+```
```
Comment (usman-jamil) [inline _posts/2024-14-18-asp-react-docker-compose.md:100]: Remove this, we will have a separate blog for docker installation

File: _posts/2024-14-18-asp-react-docker-compose.md [hunk 15]
```diff
@@ -0,0 +1,250 @@
...
+    text-align: left;
+    margin: 12px 0;
+    text-align: center;
+    font-size: 16px;
+  }
+
+---
+
+ASP Dot Net 8.0 and React.ts with Docker compose
+Create root folder name like `asp-react-nginx-compose` or other suit able.
```
Comment (usman-jamil) [inline _posts/2024-14-18-asp-react-docker-compose.md:43]: suitable is one word.

File: _posts/2024-14-18-asp-react-docker-compose.md (hunk 12 above)
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:60]: removed

File: _posts/2024-14-18-asp-react-docker-compose.md (hunk 13 above)
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:63]: done

File: _posts/2024-14-18-asp-react-docker-compose.md (hunk 15 above)
Comment (Luqmant51) [reply _posts/2024-14-18-asp-react-docker-compose.md:43]: suitable

**Review by ZuhaibSohail163 — APPROVED** [review]

**Pull request merged**